src/Data/RDF/State.hs

```diff
@@ -0,0 +1,97 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Data.RDF.State where
+
+import           Control.Monad.State
+import           Data.RDF            (RDF, Rdf, PrefixMappings, BaseUrl, Triples, Triple, NodeSelector)
+import qualified Data.RDF            as RDF
+
+-- | StateT monad which allows to combine more easily functions from 'Rdf'
+-- typeclass, eg.:
+--
+-- @
+-- import Data.RDF
+-- import qualified Data.RDF.State as RDFS
+--
+-- main :: IO ()
+-- main = do
+--   let myEmptyGraph = empty :: RDF TList
+--   newGraph <- execStateT createGraph myEmptyGraph
+--   putStrLn (showGraph newGraph)
+--
+-- createGraph :: (Rdf rdfImpl, Monad m) => RdfST rdfImpl m ()
+-- createGraph = do
+--   -- add a triple to the empty graph
+--   let triple1 = triple (unode "http://www.example.com/rob")
+--                        (unode "http://xmlns.com/foaf/0.1/interest")
+--                        (unode "http://dbpedia.org/resource/Scotch_whisky")
+--   RDFS.addTriple triple1
+--
+--   -- add another triple to the graph
+--   let triple2 = triple (unode "http://www.example.com/rob")
+--                        (unode "http://xmlns.com/foaf/0.1/interest")
+--                        (unode "http://dbpedia.org/resource/Haskell_(programming_language)")
+--   RDFS.addTriple triple2
+--
+--   -- remove one of my interests
+--   RDFS.removeTriple triple1
+-- @
+type RdfST rdfImpl m a = StateT (RDF rdfImpl) m a
```

Comment:
I'm on the fence about this idea but we might consider using a `newtype` wrapper here to hide use of `StateT`.

Reply:
Right! I'll use a newtype instead

Reply:
@tmciver Why are you on the fence exactly?

Reply:
Only because it doesn't buy a whole lot other than hiding the use of `StateT`.  I think that's a good idea and I would use a `newtype` myself but it's hard to justify without a clear benefit.  It was more an acknowledgement that not everyone would agree with this.

Reply:
Just converted to `newtype`. Is that as expected?